### DISCUSSION TITLE
Request for clearer error troubleshooting since introduction of TwillCapsules

### DISCUSSION THREAD

**Discussion by adam-jones-net** [area17/twill#2158] [Q&A]
I've come to realise as I've duplicated some of my modules in a project recently that twill often throws up rather unhelpful errors due to exceptions easily occurring in the TwillCapsules.php file.

I will give you three examples of different errors on my part that all resulted with the exception occurring in this file, which was really misleading as I'm not using Capsules in this project, so the exception resulted in me hunting around a lot of things that turned out to not bare much relevance on the actual issue.

I just setup a new module and ended up with a /app/Repositories file that had the wrong file name.  As a result I got the following error: 

```
A17\Twill\Exceptions\ : NoCapsuleFoundException
in [\vendor\area17\twill\src\TwillCapsules.php ](http://localhost/admin/properties/features)(line 74)
```

I got the same error if I set a request file named incorrectly in the /Http/Requests/Admin/ folder.
Also the same error occurs if the file was named correctly but the class name declared in the file had a typo.

I don't have a solution to this but I hope you can see that these unhandled exceptions, that admittedly occurred due to random user error are quite hard to debug.  Hoping there is a way to make this clearer in the future.

**Comment by haringsrob** [comment]
This happens mainly because the capsule check is the last one in the list. I will check what we can do about this.

**Reply by adam-jones-net** [reply]
> This happens mainly because the capsule check is the last one in the list. I will check what we can do about this.

Any news on this @haringsrob ?   I reckon its a common issue for people that upgraded from v2 to v3.  I'm still in this process because I put it off as long as I could due to seeing how much there was to do the first time I attempted a v3 upgrade.   Now that Laravel 10 is EOL and Laravel 11 isn't compatible with twill 2 it's forced me to confront the Twill upgrade again but I'm stuck on this.
